### PR TITLE
Fix offline

### LINKED ISSUE #3750: Digest cycle conflicts when offline
Offline does not work anymore

See description in:

https://forum.camptocamp.org/t/probleme-site-web-map-geo-admin-ch-sur-iphone/182512/5

### PR DESCRIPTION
**Pull request by oterral**
[test](https://mf-geoadmin3.int.bgdi.ch/teo_fixes/index.html?lang=fr&topic=ech&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege&layers_visibility=false,false,false,false&layers_timestamp=18641231,,,&X=199964.47&Y=650035.53&zoom=1)

Fix #3750

### CODE REVIEW

**Comment by davidoesch** [comment]
Ok tested on Android 7.1.1 latest chrome. Offline works

**Comment by cedricmoullet** [comment]
Tested on iPhone 6: ok
Tested on Chrome Mac: ok
Tested on Safari Mac: ok

**Comment by oterral** [comment]
The user concerned has validated the fix.

**Comment by loicgasser** [comment]
@oterral maybe you should consider using 

$httpProvider.useApplyAsync(true);

in the app config object.

See:
https://code.angularjs.org/1.3.8/docs/api/ng/provider/$httpProvider#useApplyAsync

A lot less code.

**Comment by oterral** [comment]
Yes there is already an issue for that. I've tested once but it brokes some stuff, I don't remember which .

File: test/specs/map/MapService.spec.js [hunk 1]
```diff
@@ -2035,6 +2036,7 @@ describe('ga_map_service', function() {
       });
     });
 
+    /* TODO: fix this test
```
Comment (loicgasser) [inline test/specs/map/MapService.spec.js]: I tried to fix this one but no luck ($timeout.flush, $rootScope.digest etc..) It works fine in fact, but the promise never resolves in the tests.

**Comment by loicgasser** [comment]
@oterral @gjn Shall we merge this for tomorrow ?

**Comment by oterral** [comment]
yes :)

**Comment by gjn** [comment]
Many components affected. So this needs good testing.

@Ariadnigaki @Sgachet Can you test this a bit in detail. One browser is enough. Test the following components in detail:

- offline
- context popup
- feedback (test different combinations)
- geolocation
- kml import
- map manipulations (zoom, pan, etc)
- profile
- query tool
- share tool
- time slider

**Comment by Sgachet** [comment]
Tested on IE Edge : ok !

**Comment by Ariadnigaki** [comment]
Tested on Chrome: ok

**Comment by gjn** [comment]
Thanks alot, Ladies.

**Comment by loicgasser** [comment]
Android: chrome ok
Unbuntu: Chromium ok
Unbuntu: Firefox ok

**Comment by gjn** [comment]
Ok. Thanks alot everybody. I'll merge and emergency-deploy this.

**Comment by gjn** [comment]
We are live with this.

**Pull request merged**